re.c

```diff
@@ -220,11 +220,16 @@ rb_memsearch_qs_utf8(const unsigned char *xs, long m, const unsigned char *ys, l
     return -1;
 }
 
+enum char_size
+{
+    WCHAR_SIZE = 2,
+    QCHAR_SIZE = 4
+};
+
 static inline long
-rb_memsearch_wchar(const unsigned char *xs, long m, const unsigned char *ys, long n)
+rb_memsearch_with_char_size(const unsigned char *xs, long m, const unsigned char *ys, long n, enum char_size char_size)
```

Comment:
As the `char_size` was just intended a local constant, you don't have to keep them.
Just literals would be enough.
```suggestion
rb_memsearch_with_char_size(const unsigned char *xs, long m, const unsigned char *ys, long n, int char_size)
```